day01/day01_v2.py

```diff
@@ -0,0 +1,37 @@
+
+from typing import List;
+import os;
+import time;
+import numpy as np
+
+def read_input(file_name):
+    current_path = os.path.dirname(__file__)
+    lines : List[int] = []
+    try:
+        with open(os.path.join(current_path, file_name), 'r') as file:
+            for line in file:
+                lines.append(int(line.strip()))  
+    except Exception as e:
+        print(f"An error occurred: {e}")
+    return lines
+
+def freq_calculations(inputs):
+    cumlsum_inputs = np.cumsum(inputs)
+    print("The sum of all frequencies is ", cumlsum_inputs[-1])
+    cumlsum_dict = {}
+    while True:
+
+        i = 0
+        for line in cumlsum_inputs:
+            i += 1
+            if line in cumlsum_dict:
+                print("The first repeating frequency is ", line)
+                print(cumlsum_inputs[i])
+                return
+            else:
+                cumlsum_dict[line] = None
+        cumlsum_inputs += cumlsum_inputs[-1]
```

Comment:
:zap: :zap: :zap: this is where something goes wrong
your idea is definetely interesting and I almost missed the importance of htis line :smile: 

Remember, the cumsum-array looks like this:
for a given array ` [1,2,3]` the cumsum array looks like this:
`[ 1, (1 + 2), (1 + 2 + 3) ] `  which is equal to `[ 1, 3, 6 ]`

Try to visualize what this line actually does and then try to think about what you actually need for the 2nd run over the `cumlsum_inputs` list

Reply:
but cool that you followed a not-so-obvious idea here! :partying_face: